### ISSUE TITLE
API to get list of supported devices

### ISSUE THREAD

**Issue by glennmatthews** [ncclient/ncclient#201]
It would be very helpful to have an API that returns the list of supported device name strings. For bonus points, an API that gets these paired with meaningful labels would be great as well. Something like:

```python
>>> ncclient.supported_devices()
['alu', 'csr', 'default', 'h3c', 'hpcomware', 'huawei', 'iosxe', 'iosxr', 'junos', 'nexus']
>>> ncclient.supported_device_labels()
{'csr': "Cisco CSR 1000V", 'hpcomware': 'HP Comware', 'nexus': "Cisco Nexus platform (NX-OS)", ...}
```

Right now I can manually build the supported devices list by inspection, but it's gross:

```python
import pkgutil
import ncclient.devices
supported_devices = list(modinfo[1] for modinfo in pkgutil.walk_packages(ncclient.devices.__path__))
```

**Comment by einarnn** [comment]
Glenn,

I don't disagree with the request as such, as it would help people see what devices there are default settings for etc., but just to be clear, even if a device doesn't appear on such a list, an RFC-compliant platform will still function correctly.

Out of interest, can I ask what you want to use this list for?

And if you have a suggested implementation, feel free to send a PR! 😉

Cheers,

Einar

**Comment by fredgan** [comment]
Hi @einarnn ,
PR #407 is merged, so this issue can be closed :)

**Comment by einarnn** [comment]
Resolved by #407
